### PR TITLE
Migrate from nose to pytest

### PR DESCRIPTION
**Pull request by ghassanmas**
Fixes (Partially) : https://github.com/edx/upgrades/issues/68

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ghassanmas! I've created [OSPR-6010](https://openedx.atlassian.net/browse/OSPR-6010) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: invideoquiz/settings.py [hunk 1]
```diff
@@ -9,10 +9,12 @@
     },
 }
 
-TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
+# No longer needs to override it (Using pytest)
+# TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
```
Comment (stvstnfrd) [inline invideoquiz/settings.py:13]: If we no longer need this, we should delete it instead of commenting it out.

File: invideoquiz/settings.py [hunk 2]
```diff
@@ -9,10 +9,12 @@
...
 
+''' 
 INSTALLED_APPS = (
-    'django_nose',
+    '',
 )
-
+'''
```
Comment (stvstnfrd) [inline invideoquiz/settings.py:19]: If we no longer need this, we should delete it instead of commenting it out.

**Review by stvstnfrd — CHANGES_REQUESTED** [review]
@ghassanmas Here are some quick thoughts. 

Note that tests are failing, so we'll need to sort that out ;)

Also, I upgraded this repo to Github Actions (from Travis-CI) this morning, so that should get picked up when you rebase/push again.

**Comment by stvstnfrd** [comment]
@ghassanmas It looks like there are also references to `nose` inside `setup.cfg`;
we should remove those too.

**Comment by ghassanmas** [comment]
@stvstnfrd I have resolved your comments, and sycned/merged with your latest push as well. 
So now the siuation is, tox is falling, but before digging into the problem should we first make sure that the `envlist` that tox needs to check against are correct?

**Comment by ghassanmas** [comment]
The `tox` tests fails except for qulaity, because there are currently no tests to run, when `pytest` runs 0 tests, it exit with code **5**, [ref doc](https://docs.pytest.org/en/6.2.x/usage.html#possible-exit-codes) . Sample error output when running `tox` in my machine: 
```
ERROR: InvocationError for command /xblock-in-video-quiz/.tox/py38-django30/bin/pytest (exited with code 5)
```
So to move forward I suppose we would either try to do hacky thing, [check this](https://github.com/pytest-dev/pytest/issues/2393) or run an actual tests!

**Comment by natabene** [comment]
@ghassanmas Thank you for your contribution. @stvstnfrd Do you plan to review this or should I reach out to Sarina about assigning this one?

**Comment by ghassanmas** [comment]
My pleasure @natabene :) 
@stvstnfrd I have just added a test case which tests for the Xblock initial deafult field value, that being said, now running  `tox` shuold succeed

File: invideoquiz/tests/test_invideoquiz_view.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from xblock.runtime import DictKeyValueStore, KvsFieldData
+from invideoquiz import InVideoQuizXBlock
+from unittest.mock import Mock
+
+
+def test_invideoquiz_view():
+    key_store = DictKeyValueStore()
+    field_data = KvsFieldData(key_store)
+    runtime = Runtime(services={'field-data': field_data})
+    instace = InVideoQuizXBlock(runtime, scope_ids=Mock())
```
Comment (sarina) [inline invideoquiz/tests/test_invideoquiz_view.py:11]: nit: could this variable be named `instance`?

File: invideoquiz/tests/test_invideoquiz_view.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from xblock.test.tools import TestRuntime as Runtime
+from xblock.runtime import DictKeyValueStore, KvsFieldData
+from invideoquiz import InVideoQuizXBlock
+from unittest.mock import Mock
+
+
+def test_invideoquiz_view():
```
Comment (sarina) [inline invideoquiz/tests/test_invideoquiz_view.py:7]: nit: could you add a docstring that explains what is being tested?

**Review by sarina — CHANGES_REQUESTED** [review]
This looks good to me - left a few nits I'd like addressed. Could you also please squash your commits and use conventional commits? You'll also need to rebase your branch.

File: invideoquiz/tests/test_invideoquiz_view.py (hunk 3 above)
Comment (stvstnfrd) [reply invideoquiz/tests/test_invideoquiz_view.py:11]: I'd second this request, and even call it _not_ a "nit"; misspelled words/variables should always be corrected ("nit" implies an optional request)

**Review by stvstnfrd — CHANGES_REQUESTED** [review]
This generally looks good and I second @sarina 's comments.

Since the previous (empty) test files are no longer applicable, would you mind removing them?
```sh
git rm invideoquiz/tests.py invideoquiz/tests/test_travis.py
```

**Comment by sarina** [comment]
@ghassanmas this now looks good to me but I cannot merge until you resolve conflicts by rebasing, and you squash your commits and use conventional commits. https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

If you need help with these, please let us know.

**Comment by sarina** [comment]
@ghassanmas Instructions to *rebase* and *squash* are here: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by ghassanmas** [comment]
@sarina Thank you for the info, took little of time to get my head around rebase, gonna keep rewriting history from now on! 😁

**Review by stvstnfrd — APPROVED** [review]
+1 from me, thanks @ghassanmas !

**Comment by edx-webhook** [comment]
@ghassanmas 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-in-video-quiz&entry.867055334=https://github.com/edx/xblock-in-video-quiz/pull/24&entry.1484655318=https://github.com/ghassanmas&entry.752974735=2021-09-01+15:13&entry.1917517419=2021-09-10+19:23&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**